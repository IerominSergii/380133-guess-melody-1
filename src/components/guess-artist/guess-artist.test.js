import React from "react";
import renderer from "react-test-renderer";
import GuessArtist from "./guess-artist";

it(`should render GuessArtist properly`, () => {
  const onAnswer = jest.fn();
  const question = {
    type: `artist`,
    song: {
      artist: `Jim Beam`,
      src: `path.mp3`
    },
    answers: [
      {
        picture: `path`,
        artist: `John Snow`
      }
    ]
  };

  const componentGuessArtist = renderer
    .create(<GuessArtist question={question} onAnswer={onAnswer} />, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(componentGuessArtist).toMatchSnapshot();
});
