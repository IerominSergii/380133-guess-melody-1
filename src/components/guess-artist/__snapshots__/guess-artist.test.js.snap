// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GuessArtist properly 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    &lt;svg
          xmlns="http://www.w3.org/2000/svg"
          className="timer"
          viewBox="0 0 780 780"
        &gt;
          &lt;circle
            className="timer__line"
            cx="390"
            cy="390"
            r="370"
            style="filter: url(#blur); transform: rotate(-90deg) scaleY(-1); transform-origin: center"
          /&gt;
        &lt;/svg&gt;
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <audio />
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          onClick={[Function]}
          type="radio"
          value="John Snow"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="John Snow"
            className="artist__picture"
            src="path"
          />
          John Snow
        </label>
      </div>
    </form>
  </section>
</section>
`;
